src/lib.rs

```diff
@@ -206,12 +214,16 @@ impl Wal {
         let mut segment = self.creator.next()?;
         mem::swap(&mut self.open_segment, &mut segment);
 
-        self.flush = Some(
-            self.flush
-                .take()
-                .unwrap_or_else(|| Future::of(()))
-                .and(segment.segment.flush_async()),
-        );
+        // run current flush
+        if let Some(flush) = self.flush.take() {
+            if flush.is_finished() {
```

Comment:
this logic is not equivalent to the previous one, where a new flush would always be schedule using `and`